### PR TITLE
feat: Add Parse copy-pasted secrets in the dashboard

### LINKED ISSUE #231: Parse copy-pasted secrets into the dashboard
### Feature description
If a user pastes `SECRET_X=VALUE_Y` into the dashabord, we should immediately split the key and the values into two input fields. 

### Why would it be useful?
This a QoL feature and it would be dramatically improve the user experience for many users who paste the environment variables / secrets into the dashaboard.

### PR DESCRIPTION
**Pull request by alceil**
# Description 📣

Feature to Parse copy-pasted secrets in the dashboard
Resolves #231 

[screen-capture.webm](https://github.com/Infisical/infisical/assets/47685349/df257771-dbc6-45d8-af62-cceb47b2aa66)

## Type ✨

- [x] New feature

# Tests 🛠️

```sh
# Here's some code block to paste some code snippets
```
- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/code-of-conduct). 📝

### CODE REVIEW

File: frontend/src/views/DashboardPage/components/SecretInputRow/SecretInputRow.tsx [hunk 1]
```diff
@@ -199,6 +199,15 @@ export const SecretInputRow = memo(
 
     const isCreatedSecret = !secId;
     const shouldBeBlockedInAddOnly = !isCreatedSecret && isAddOnly;
+    
+    const hasEqualSign = (str:string) => str.includes("=");
```
Comment (akhilmhdh) [inline frontend/src/views/DashboardPage/components/SecretInputRow/SecretInputRow.tsx:203]: can you keep this function outside the component if its just a utility function

File: frontend/src/views/DashboardPage/components/SecretInputRow/SecretInputRow.tsx [hunk 2]
```diff
@@ -245,6 +254,9 @@ export const SecretInputRow = memo(
                         isKeySubDisabled.current = false;
                         field.onBlur();
                       }}
+                      onChange={(val)=>{
+                        onChange(hasEqualSign(val.target.value)?splitKeyValue(val.target.value):val);
```
Comment (akhilmhdh) [inline frontend/src/views/DashboardPage/components/SecretInputRow/SecretInputRow.tsx:258]: The issue with this logic is whenever secret value has = it gets split into two parts which can confuse user. 

This should happen only when user is pasting into the input field.

Comment (akhilmhdh) [reply frontend/src/views/DashboardPage/components/SecretInputRow/SecretInputRow.tsx:258]: Also are you sure value is getting saved, just asking to be sure. Coz the editorRef change may be affected only in ui not in state

File: frontend/src/views/DashboardPage/components/SecretInputRow/SecretInputRow.tsx (hunk 1 above)
Comment (alceil) [reply frontend/src/views/DashboardPage/components/SecretInputRow/SecretInputRow.tsx:203]: sure @akhilmhdh

**Comment by alceil** [comment]
I have added some relevant changes can you please have a look👀 @akhilmhdh

File: frontend/src/views/DashboardPage/components/SecretInputRow/SecretInputRow.tsx [hunk 3]
```diff
@@ -218,6 +225,15 @@ export const SecretInputRow = memo(
       return <></>;
     }
 
+    let isPasted = false; 
+    
+    const handlePaste = () => {
```
Comment (akhilmhdh) [inline frontend/src/views/DashboardPage/components/SecretInputRow/SecretInputRow.tsx:230]: Using `onPaste` event for this is indeed a good move. 

But using this way makes it so complicated. I would request you to revert back the onchange things and instead here you will get the event and then using that event u will get clipboard content. Using `setValue` function from react hook form and setEditor ref set the things accordingly

File: frontend/src/views/DashboardPage/components/SecretInputRow/SecretInputRow.tsx [hunk 4]
```diff
@@ -218,6 +225,15 @@ export const SecretInputRow = memo(
       return <></>;
     }
 
+    let isPasted = false;
```
Comment (akhilmhdh) [inline frontend/src/views/DashboardPage/components/SecretInputRow/SecretInputRow.tsx:228]: Never put state change like this its a bad practice

**Review by akhilmhdh — CHANGES_REQUESTED** [review]

**Comment by maidul98** [comment]
Changes have not been be addressed. Closing for now

**Pull request closed without merging**